src/node/filesystem/binary.node.test.ts

```diff
@@ -0,0 +1,293 @@
+      expect(read).toStrictEqual(Buffer.from('12345678910'));
+    });
+
+    it('pauses stream after readding byte', async () => {
+      const pauseSpy = jest.spyOn(stream, 'pause');
+  
+      await reader.read();
+      expect(pauseSpy).toHaveBeenCalledTimes(1)  
+    });
+  });
+
+  describe('RedableClone class', () => {
+    it('gets same data through both clones', async () => {
+      const original = new MockStream();
+      const firstClone = new ReadableClone(original);
+      const secondClone = new ReadableClone(original);
+
+      const firstChunks = [];
+      for await (const chunk of firstClone) {
+        firstChunks.push(chunk);
+      }
+
+      const secondChunks = [];
+      for await (const chunk of secondClone) {
+        secondChunks.push(chunk);
+      }
+
+      expect(firstChunks).toEqual(secondChunks);
+    });
+  });
+
+  describe('FileContainer class', () => {
+    let fileContainer: FileContainer;
+
+    beforeEach(async () => {
+      fileContainer = new FileContainer(fixturePath, { filename: 'testfile.txt', mimetype: 'text/plain' });
+      await fileContainer.initialize();
+    });
+
+    afterEach(async () => {
+      await fileContainer.destroy();
+    });
+
+    it('opens file correctly', async () => {
+      expect(fileContainer.filesize).toBeGreaterThan(0);
+    });
+
+    it('throws an error if the file does not exist', async () => {
+      const fileContainer = new FileContainer('non-existent.txt');
+      await expect(fileContainer.initialize()).rejects.toThrow(NotFoundError);
+    });
+
+    it('closes handle and resets data container', async () => {
+      const closeHandleSpy = jest.spyOn((fileContainer as any).handle, 'close');
+
+      await fileContainer.destroy();
+
+      expect(closeHandleSpy).toHaveBeenCalled();
+      expect(fileContainer.filesize).toBe(Infinity);
+      expect((fileContainer as any).handle).toBe(undefined);
+      expect((fileContainer as any).stream).toBe(undefined);
+      expect((fileContainer as any).streamReader).toBe(undefined);
+    });
+
+    it('sets filesize', async () => {
+      expect(fileContainer.filesize).not.toBe(Infinity);
+    });
+
+    it('creates ReadableClone for data stream', async () => {
+      expect((fileContainer as any).stream).toBeInstanceOf(ReadableClone);
+    });
+
+    it('passes ReadableClone to StreamReader', async () => {
+      expect((fileContainer as any).streamReader.stream).toBeInstanceOf(ReadableClone);
+    });
+  });
+
+  describe('StreamContainer class', () => {
+    let streamContainer: StreamContainer;
+
+    beforeEach(async () => {
+      streamContainer = new StreamContainer(new MockStream());
+    });
+
+    it('creates ReadableClone for data stream', async () => {
+      expect((streamContainer as any).stream).toBeInstanceOf(ReadableClone);
+    });
+
+    it('passes ReadableClone to StreamReader', async () => {
+      expect((streamContainer as any).streamReader.stream).toBeInstanceOf(ReadableClone);
+    });
```

Comment:
question: should we really be testing for this? Seems like a very deep implementation detail to me.

Reply:
Outdated